src/veh_interact.cpp

```diff
@@ -1569,8 +1564,8 @@ void veh_interact::display_stats()
                    _("Acceleration: <color_ltblue>%3d</color> %s/t"),
                    int(veh->acceleration(false) * speed_factor), speed_units.c_str());
     fold_and_print(w_stats, y[2], x[2], w[2], c_ltgray,
-                   _("Mass: <color_ltblue>%5d</color> %s"),
-                   int(veh->total_mass() * weight_factor), weight_units.c_str());
+                   _("Mass: <color_ltblue>%5.0f</color> %s"),
+                   convert_weight(veh->total_mass() * 1000.0f), weight_units().c_str());
```

Comment:
I was too lazy for removing ```/1000``` from return of ```total_mass()``` function and updating all places where this function is used, so I've just put the additional multiplier here. Can be managed later (in separate PR, if it worth the efforts).